Backport of

commit 569beb73e9d5cca4d567388dd297eb7fbbc3df42
  ARM: OMAP3: Adding vdd_sdi regulator supply to IGEPv2 board
commit 903ae55218669baf230c765be52aeb5dbd218bf0
  ARM: OMAP3: Add S-Video output to IGEPv2 board

from tree
http://kernel.ubuntu.com/git?p=rsalveti/ubuntu-maverick.git;a=shortlog;h=refs/heads/fix-igepv2-display

Refer to https://bugs.launchpad.net/ubuntu/maverick/+source/linux/+bug/607250
for details.

diff -Naur linux-2.6.38-rc4/arch/arm/mach-omap2/board-igep0020.c linux-2.6.38-rc4a/arch/arm/mach-omap2/board-igep0020.c
--- linux-2.6.38-rc4/arch/arm/mach-omap2/board-igep0020.c	2011-02-08 01:03:55.000000000 +0100
+++ linux-2.6.38-rc4a/arch/arm/mach-omap2/board-igep0020.c	2011-02-13 16:05:59.958506028 +0100
@@ -475,8 +475,16 @@
 	.phy.dpi.data_lines	= 24,
 };
 
+static struct omap_dss_device igep2_tv_device = {
+	.name			= "tv",
+	.driver_name		= "venc",
+	.type			= OMAP_DISPLAY_TYPE_VENC,
+	.phy.venc.type		= OMAP_DSS_VENC_TYPE_SVIDEO,
+};
+
 static struct omap_dss_device *igep2_dss_devices[] = {
-	&igep2_dvi_device
+	&igep2_dvi_device,
+	&igep2_tv_device,
 };
 
 static struct omap_dss_board_info igep2_dss_data = {
@@ -493,9 +501,29 @@
 	},
 };
 
-static struct regulator_consumer_supply igep2_vpll2_supply = {
-	.supply	= "vdds_dsi",
-	.dev	= &igep2_dss_device.dev,
+static struct regulator_consumer_supply igep2_vdda_dac_supply = {
+	.supply = "vdda_dac",
+	.dev    = &igep2_dss_device.dev,
+};
+
+/* VDAC for DSS driving S-Video (8 mA unloaded, max 65 mA) */
+static struct regulator_init_data igep2_vdac = {
+	.constraints = {
+		.min_uV			= 1800000,
+		.max_uV			= 1800000,
+		.valid_modes_mask	= REGULATOR_MODE_NORMAL
+					| REGULATOR_MODE_STANDBY,
+		.valid_ops_mask		= REGULATOR_CHANGE_MODE
+					| REGULATOR_CHANGE_STATUS,
+	},
+	.num_consumer_supplies	= 1,
+	.consumer_supplies	= &igep2_vdda_dac_supply,
+};
+
+/* VPLL2 for digital video outputs */
+static struct regulator_consumer_supply igep2_vdds_supplies[] = {
+	REGULATOR_SUPPLY("vdds_sdi", "omapdss"),
+	REGULATOR_SUPPLY("vdds_dsi", "omapdss"),
 };
 
 static struct regulator_init_data igep2_vpll2 = {
@@ -509,8 +537,8 @@
 		.valid_ops_mask		= REGULATOR_CHANGE_MODE
 					| REGULATOR_CHANGE_STATUS,
 	},
-	.num_consumer_supplies	= 1,
-	.consumer_supplies	= &igep2_vpll2_supply,
+	.num_consumer_supplies	= ARRAY_SIZE(igep2_vdds_supplies),
+	.consumer_supplies	= igep2_vdds_supplies,
 };
 
 static void __init igep2_display_init(void)
@@ -584,6 +612,7 @@
 	.keypad		= &igep2_keypad_pdata,
 	.vmmc1          = &igep2_vmmc1,
 	.vpll2		= &igep2_vpll2,
+	.vdac		= &igep2_vdac,
 	.vio		= &igep2_vio,
 };
 
